#Bài tập 1: Tải và Chuyển đổi Dữ liệu Iris


##Yêu cầu 1.1: Tải dữ liệu



In [5]:
# Them cac thu vien can thiet
import pandas as pd
from sklearn.datasets import load_iris

In [6]:
iris_data = load_iris()
iris_data.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

##Yêu cầu 1.2: Tạo DataFrame ban đầu



In [7]:
df = pd.DataFrame(data=iris_data.data, columns=iris_data.feature_names)
df

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                  5.1               3.5                1.4               0.2
1                  4.9               3.0                1.4               0.2
2                  4.7               3.2                1.3               0.2
3                  4.6               3.1                1.5               0.2
4                  5.0               3.6                1.4               0.2
..                 ...               ...                ...               ...
145                6.7               3.0                5.2               2.3
146                6.3               2.5                5.0               1.9
147                6.5               3.0                5.2               2.0
148                6.2               3.4                5.4               2.3
149                5.9               3.0                5.1               1.8

[150 rows x 4 columns]

##Yêu cầu 1.3: Thêm thông tin về loài hoa (Target)


In [8]:
df['species'] = iris_data.target
df

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                  5.1               3.5                1.4               0.2   
1                  4.9               3.0                1.4               0.2   
2                  4.7               3.2                1.3               0.2   
3                  4.6               3.1                1.5               0.2   
4                  5.0               3.6                1.4               0.2   
..                 ...               ...                ...               ...   
145                6.7               3.0                5.2               2.3   
146                6.3               2.5                5.0               1.9   
147                6.5               3.0                5.2               2.0   
148                6.2               3.4                5.4               2.3   
149                5.9               3.0                5.1               1.8   

     species  
0          0  
1          0  
2          0  
3          0  
4          0  
..       ...  
145        2  
146        2  
147        2  
148        2  
149        2  

[150 rows x 5 columns]

#Bài tập 2: Khám phá và Phân tích Dữ liệu Iris


##Yêu cầu 2.1: Khám phá tổng quan


In [9]:
df.shape

(150, 5)

Bộ dữ liệu có chính xác 150 dòng và 5 cột.
Bộ dữu liệu không có giá trị nào bị null.
Kiểu dữ liệu của các cột đặc trưng là int, float64

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 6.0 KB


##Yêu cầu 2.2: Thống kê mô tả

Đặc trưng sepal length có mean lớn nhất 5.843333.
Đặc trưng có độ lệch chuẩn lớn nhất std là petal length 1.765298. Điều này cho thấy sự phân bố của các cột xa hơn so với giá trị trung bình


In [12]:
df.describe()

sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)     species  
count        150.000000  150.000000  
mean           1.199333    1.000000  
std            0.762238    0.819232  
min            0.100000    0.000000  
25%            0.300000    0.000000  
50%            1.300000    1.000000  
75%            1.800000    2.000000  
max            2.500000    2.000000

##Yêu cầu 2.3: Phân tích theo loài hoa



Mỗi loài hoa có 50 mẫu. Dữ liệu này cân bằng

In [13]:
df['species'].value_counts()

species
0    50
1    50
2    50
Name: count, dtype: int64

In [14]:
iris_data.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

Virginica là loài hoa có petal length trung bình lớn nhất.
Setosa là loài hoa có sepal width trung bình lớn nhất



In [15]:
df.groupby('species').mean()

sepal length (cm)  sepal width (cm)  petal length (cm)  \
species                                                           
0                    5.006             3.428              1.462   
1                    5.936             2.770              4.260   
2                    6.588             2.974              5.552   

         petal width (cm)  
species                    
0                   0.246  
1                   1.326  
2                   2.026

#Bài tập 3: Thu thập và Khám phá dữ liệu Điểm thi THPT 2024


##Yêu cầu 3.0: Đọc dữ liệu từ file CSV cục bộ


In [19]:
import pandas as pd
url = "/content/drive/MyDrive/Colab Notebooks/Data/sinhvien.csv"

df_demo = pd.read_csv(url)
df_demo

mssv        ho_ten chuyen_nganh  diem_trung_binh
0  22120001  Nguyen Van A         KHMT              8.5
1  22120002    Tran Thi B         HTTT              7.9
2  22120003      Le Van C         CNPM              8.1

##Yêu cầu 3.1: Tải dữ liệu từ URL


In [20]:
import pandas as pd
url_diem_thi = "https://github.com/anhdung98/diem_thi_2024/releases/download/240717/diem_thi_thpt_2024.csv"

df_diem_thi = pd.read_csv(url_diem_thi)
df_diem_thi.head(5)

sbd  toan  ngu_van  ngoai_ngu  vat_li  hoa_hoc  sinh_hoc  lich_su  \
0  1000001   8.4     6.75        8.0     6.0     5.25       5.0      NaN   
1  1000002   8.6     8.50        7.2     NaN      NaN       NaN     7.25   
2  1000003   8.2     8.75        8.2     NaN      NaN       NaN     7.25   
3  1000004   4.8     8.50        7.4     NaN      NaN       NaN     7.00   
4  1000005   8.6     9.00        7.8     NaN      NaN       NaN     9.00   

   dia_li  gdcd ma_ngoai_ngu  
0     NaN   NaN           N1  
1    6.00  8.00           N1  
2    7.25  8.75           N1  
3    6.00  7.50           N1  
4    8.75  8.50           N1

##Yêu cầu 3.2: Khám phá tổng quan


Bộ dữ liệu điểm thi THPT 2024 có chính xác 1061605 dòng và 11 cột

In [21]:
df_diem_thi.shape

(1061605, 11)

Các cột có trong dữ liệu: 'sbd', 'toan', 'ngu_van', 'ngoai_ngu', 'vat_li', 'hoa_hoc', 'sinh_hoc',
       'lich_su', 'dia_li', 'gdcd', 'ma_ngoai_ngu'

In [22]:
df_diem_thi.columns

Index(['sbd', 'toan', 'ngu_van', 'ngoai_ngu', 'vat_li', 'hoa_hoc', 'sinh_hoc',
       'lich_su', 'dia_li', 'gdcd', 'ma_ngoai_ngu'],
      dtype='object')

Loại dữ liệu của cột sbd là int64

Cột Toan có 1045613 giá trị không rỗng. Điều này nói lên có một vài thí sinh bỏ thi môn toán

In [23]:
df_diem_thi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1061605 entries, 0 to 1061604
Data columns (total 11 columns):
 #   Column        Non-Null Count    Dtype  
---  ------        --------------    -----  
 0   sbd           1061605 non-null  int64  
 1   toan          1045613 non-null  float64
 2   ngu_van       1050101 non-null  float64
 3   ngoai_ngu     912705 non-null   float64
 4   vat_li        345615 non-null   float64
 5   hoa_hoc       346518 non-null   float64
 6   sinh_hoc      342378 non-null   float64
 7   lich_su       706214 non-null   float64
 8   dia_li        704682 non-null   float64
 9   gdcd          583609 non-null   float64
 10  ma_ngoai_ngu  912705 non-null   object 
dtypes: float64(9), int64(1), object(1)
memory usage: 89.1+ MB


##Yêu cầu 3.3: Thống kê mô tả cơ bản


Điểm trung bình của môn Văn là 7.23

Thí sinh có điểm môn lịch sử cao nhất là 10 điểm

Quan sát giá trị nhỏ nhất (min) của các cột điểm. Ta thấy môn nào điểm thấp nhất là 0 điểm. Điều này cho thấy thí sinh quên thi, nộp giấy trắng, sai cú pháp

In [24]:
df_diem_thi.describe()

sbd          toan       ngu_van      ngoai_ngu         vat_li  \
count  1.061605e+06  1.045613e+06  1.050101e+06  912705.000000  345615.000000   
mean   2.734354e+07  6.447309e+00  7.231146e+00       5.521938       6.666866   
std    1.919442e+07  1.556900e+00  1.330057e+00       1.880657       1.631129   
min    1.000001e+06  0.000000e+00  0.000000e+00       0.000000       0.000000   
25%    9.006324e+06  5.400000e+00  6.500000e+00       4.000000       5.500000   
50%    2.800794e+07  6.800000e+00  7.500000e+00       5.200000       7.000000   
75%    4.400492e+07  7.600000e+00  8.250000e+00       7.000000       8.000000   
max    6.400729e+07  9.800000e+00  1.000000e+01      10.000000      10.000000   

            hoa_hoc       sinh_hoc        lich_su         dia_li  \
count  346518.00000  342378.000000  706214.000000  704682.000000   
mean        6.68083       6.284079       6.570343       7.193541   
std         1.64147       1.244849       1.462804       1.283945   
min         0.00000       0.000000       0.000000       0.000000   
25%         5.50000       5.500000       5.500000       6.500000   
50%         6.75000       6.250000       6.500000       7.250000   
75%         8.00000       7.250000       7.500000       8.000000   
max        10.00000      10.000000      10.000000      10.000000   

                gdcd  
count  583609.000000  
mean        8.156503  
std         1.042644  
min         0.000000  
25%         7.500000  
50%         8.250000  
75%         9.000000  
max        10.000000